platform/stm32f4xx/irq_manager.cpp

```diff
@@ -0,0 +1,90 @@
+#include <platform/irq_manager.hpp>
+#include <new>
+
+IRQ_manager::handler_storage IRQ_manager::m_storage[IRQ_manager::irqs];
+
+
+void IRQ_manager::init()
+{
+    // Initialize storage
+    for (auto &h : m_storage) {
+        new (&h) handler_type{default_handler};
+    }
+
+    __enable_irq();
+}
+
+int IRQ_manager::subscribe(IRQn_t irqn, const std::function< void() > &handler)
+{
```

Comment:
IRQ manager in current implementation doesn't support handling of system exceptions. To subscribe for exception events one may override [handlers](https://github.com/forGGe/theCore/blob/master/platform/stm32f4xx/startup/start.s#L80) at compile time via  mechanism: